### PR TITLE
Add scheduler UI to ballista-scheduler Docker image

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-ballista/issues/11

 # Rationale for this change

With this PR, I can connect to the scheduler web UI. The only functionality so far is being able to see a list of executors but with this in place we can iterate on adding new functionality such as viewing jobs, stages, and metrics.

# What changes are included in this PR?

- Create separate images for scheduler vs executor
- Prefix image names with `apache/arrow-`
- Include scheduler UI in scheduler image
- Use `ENTRYPOINT` instead of `CMD` inside scheduler and executor images

# Are there any user-facing changes?

No

### CODE REVIEW

File: dev/docker/ballista-scheduler.dockerfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+WORKDIR /tmp/ballista-ui
+
+COPY ballista/ui/scheduler ./
+RUN yarn
+
+# TODO this fails with:
+# Error: Cannot find module '../../data/browsers'
+#RUN yarn build
+
+# TODO start nginx in background to serve the UI
```
Comment (andygrove) [inline dev/docker/ballista-scheduler.dockerfile:35]: I need help with this part as I have zero experience with node or yarn :weary:

Comment (andygrove) [reply dev/docker/ballista-scheduler.dockerfile:35]: cc @msathis

**Comment by andygrove** [comment]
@yahoNanJing @thinkharderdev @avantgardnerio This PR is now ready for review

**Review by avantgardnerio — APPROVED** [review]
LGTM

**Review by thinkharderdev — APPROVED** [review]
Nice!

**Comment by avantgardnerio** [comment]
Pictures or it didn't happen :)

**Review by yahoNanJing — COMMENTED** [review]
LGTM

**Comment by yahoNanJing** [comment]
Maybe better to add an example command to leverage these docker files. For example, like follows:
{code}
./dev/build-ballista-docker.sh

docker run -p 10.249.75.38:80:80 -p 10.249.75.38:50050:50050 -it apache/arrow-ballista-scheduler:0.8.0 --external-host 10.249.75.38 -s push-staged

docker run -p 10.249.75.38:50051:50051 -p 10.249.75.38:50052:50052 -it apache/arrow-ballista-executor:0.8.0 --external-host 10.249.75.38 --scheduler-host 10.249.75.38 -s push-staged
{code}

And it seems we need to fix something in file *executor_server.rs* to change the following part 
{code}
        let addr = format!(
            "{}:{}",
            executor_meta
                .optional_host
                .map(|h| match h {
                    OptionalHost::Host(host) => host,
                })
                .unwrap_or_else(|| String::from("0.0.0.0")),
            executor_meta.grpc_port
        );
{code}
to
{code}
let addr = format!(
            "{}:{}",
            "0.0.0.0",
            executor_meta.grpc_port
        );
{code}

**Comment by andygrove** [comment]
> Pictures or it didn't happen :)

ballista-ui

The UI says "scheduler hasn't started" now, but I saw this working previously. Once this PR is merged, I will start learning more about the UI code and see if I can start adding the new functionality.

**Comment by andygrove** [comment]
> Maybe better to add an example command to leverage these docker files.

I have now updated the relevant sections in the user guide.

**Comment by avantgardnerio** [comment]
> add an example command

I personally favor `docker-compose.yml` because it can build the images and store all the runtime information like ports in a formal way.

**Comment by andygrove** [comment]
> > add an example command
> 
> I personally favor `docker-compose.yml` because it can build the images and store all the runtime information like ports in a formal way.

Here is the user guide documentation for using docker-compose:

https://github.com/apache/arrow-ballista/blob/master/docs/source/user-guide/deployment/docker-compose.md

**Comment by avantgardnerio** [comment]
> documentation for using docker-compose

I don't understand `./dev/build-ballista-docker.sh`? Why not simply 
```
docker-compose up --build
```
?

**Comment by andygrove** [comment]
> > documentation for using docker-compose
> 
> I don't understand `./dev/build-ballista-docker.sh`? Why not simply
> 
> ```
> docker-compose up --build
> ```
> 
> ?

I'm not familiar with that feature. Do you want to PR to add that? We will still need a way to create the images for other uses, like k8s.

**Comment by andygrove** [comment]
> Also, why is out `docker-compose.yml` inside a `readme.md` vs just checked in?

This is for the user guide that will soon be published to the Arrow site. It is common practice for user guides to have sample yaml along with an explanation (see Spark's docs for example).

Ideally, we would have the yaml checked into the repo and tested and then generate that into the docs. I would like to get to that point eventually.

We do have one tested docker-compose.yaml checked into the `benchmarks` directory, and this is tested by the integration test script.

**Comment by avantgardnerio** [comment]
I was just in the process of making PRs out of all our changes now that DataFusion 12 was released. This is the PR: https://github.com/apache/arrow-ballista/pull/249

**Comment by avantgardnerio** [comment]
We also have a helm chart I'd like to PR in as well to handle kubernetes.

**Pull request merged**